ducktape/tests/loader.py

```diff
@@ -230,10 +250,24 @@ def expand_class(self, t_info):
                 t.function =_inject(**self.test_parameters)(t.function)
                 t.injected_args = self.test_parameters
                 test_info_list.append(t)
-
+                
+        test_info_list.extend(self.expand_regressions(test_info_list))
         return test_info_list
-
-    def expand_function(self, t_info):
+    
+    def expand_regressions(self, t_info_list):
+        regression_list = []
+        
+        for t_info in t_info_list:
+            if has_regression(t_info.function):
+                for mark in t_info.function.marks:
+                    if isinstance(mark, RegressionMark):
+                        t_copy = TestInfo.from_test_info(t_info)
+                        t_copy.regression_variable_selector = mark.variable_selector
```

Comment:
The clone method could also just accept additional keyword arguments to provide overrides. That would turn this into a one-liner like

```
regression_list.append(t_info.clone(regression_variable_selector=mark.variable_selector))
```